### PR TITLE
Expose functions in storage/remote.

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

This PR exposes some go objects in `storage/remote` package so that community tools can reuse them. While developing a remote-write benchmarker (in Promscale repo), we found these would help us along with other projects in the community that are dependent on `remote` package.

### CODE REVIEW

**Comment by roidelapluie** [comment]
Side note: have you looked at https://github.com/open-fresh/avalanche ? Some members of the community are already using this and it will probably become a prometheus-community project.

**Comment by roidelapluie** [comment]
Do you still need this?

**Comment by Harkishen-Singh** [comment]
Yes merging this will help.

**Comment by roidelapluie** [comment]
Okay. There is current work to expose the interner in #10231 so It should help with part of this. Once it is merged, I will revisit this one.

**Comment by roidelapluie** [comment]
We have looked at this pull request during our bug scrub. It seems like we don't need the interner with the new stringlabels build option, so if you rebase this pull request I will merge it.

File: storage/remote/queue_manager.go [hunk 1]
```diff
@@ -366,29 +366,29 @@ type QueueManager struct {
 	quit        chan struct{}
 	wg          sync.WaitGroup
 
-	dataIn, dataDropped, dataOut, dataOutDuration *ewmaRate
+	DataIn, DataDropped, DataOut, DataOutDuration *EWMARate
```
Comment (bboreham) [inline storage/remote/queue_manager.go:369]: I think these should be made available via function, if you need them, rather than by exporting just these three members.

File: storage/remote/queue_manager.go [hunk 2]
```diff
@@ -415,17 +415,19 @@ func NewQueueManager(
...
+		DataOut:         NewEWMARate(ewmaWeight, shardUpdateDuration),
+		DataOutDuration: NewEWMARate(ewmaWeight, shardUpdateDuration),
 
 		metrics:              metrics,
 		interner:             interner,
 		highestRecvTimestamp: highestRecvTimestamp,
 	}
 
-	t.watcher = wal.NewWatcher(watcherMetrics, readerMetrics, logger, client.Name(), t, walDir, enableExemplarRemoteWrite)
+	if walDir != "" {
```
Comment (bboreham) [inline storage/remote/queue_manager.go:428]: Change not covered by commit description, seems unrelated to PR title.
Please put in a different PR with explanation.

File: storage/remote/queue_manager.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ const (
 	shardToleranceFraction = 0.3
 )
 
-type queueManagerMetrics struct {
+type QueueManagerMetrics struct {
```
Comment (bboreham) [inline storage/remote/queue_manager.go:53]: How can you use this type from outside when none of its behaviour is exported?

**Comment by roidelapluie** [comment]
As we did not have any feedback I will close this PR.
